conda_build/build.py

```diff
@@ -615,8 +616,9 @@ def build(m, config, post=None, need_source_download=True, need_reparse_in_env=F
 
                     # There is no sense in trying to run an empty build script.
                     if isfile(build_file) or script:
-                        env = environ.get_dict(config=config, m=m, dirty=config.dirty)
-                        env["CONDA_BUILD_STATE"] = "BUILD"
+                        with path_prepended(config.build_prefix):
+                            env = environ.get_dict(config=config, m=m, dirty=config.dirty)
+                            env["CONDA_BUILD_STATE"] = "BUILD"
```

Comment:
not terribly important, but I think the scope of env["CONDA_BUILD_STATE"] is one level out, since it is agnostic of the PATH.